newrelic/hooks/mlmodel_langchain.py

```diff
@@ -118,17 +122,35 @@
     "langchain_community.vectorstores.weaviate": "Weaviate",
     "langchain_community.vectorstores.xata": "XataVectorStore",
     "langchain_community.vectorstores.yellowbrick": "Yellowbrick",
+    "langchain_community.vectorstores.zep_cloud": "ZepCloudVectorStore",
     "langchain_community.vectorstores.zep": "ZepVectorStore",
     "langchain_community.vectorstores.docarray.hnsw": "DocArrayHnswSearch",
     "langchain_community.vectorstores.docarray.in_memory": "DocArrayInMemorySearch",
 }
 
 
-def _create_error_vectorstore_events(transaction, search_id, args, kwargs, linking_metadata):
+def bind_submit(func, *args, **kwargs):
+    return {"func": func, "args": args, "kwargs": kwargs}
+
+
+def wrap_ContextThreadPoolExecutor_submit(wrapped, instance, args, kwargs):
+    trace = current_trace()
+    if not trace:
+        return wrapped(*args, **kwargs)
+
+    # Use hardened function signature bind so we have safety net catchall of args and kwargs.
+    bound_args = bind_submit(*args, **kwargs)
+    bound_args["func"] = context_wrapper(bound_args["func"], trace=trace, strict=True)
+    return wrapped(bound_args["func"], *bound_args["args"], **bound_args["kwargs"])
+
+
+def _create_error_vectorstore_events(transaction, search_id, args, kwargs, linking_metadata, wrapped):
     settings = transaction.settings if transaction.settings is not None else global_settings()
     span_id = linking_metadata.get("span.id")
     trace_id = linking_metadata.get("trace.id")
-    request_query, request_k = bind_similarity_search(*args, **kwargs)
+    bound_args = bind_args(wrapped, args, kwargs)
+    request_query = bound_args["query"]
+    request_k = bound_args["k"]
```

Comment:
Then this means that the function signature has changed. I rather have it fail fast in these cases rather than silently fail without us knowing since these are arguments we depend on to instrument.